src/Sarif.Viewer.VisualStudio.Core/OutputWindowTracerListener.cs

```diff
@@ -0,0 +1,83 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Diagnostics;
+
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+
+namespace Microsoft.Sarif.Viewer
+{
+    public class OutputWindowTracerListener : TraceListener
+    {
+        private readonly string _name;
+        private readonly IVsOutputWindow _outputWindowService;
+
+        private IVsOutputWindowPane pane;
+
+        public OutputWindowTracerListener(IVsOutputWindow outputWindowService, string name)
+        {
+            this._outputWindowService = outputWindowService;
+            this._name = name;
+            Trace.Listeners.Add(this);
+        }
+
+        public override void Write(string message)
+        {
+            if (!SarifViewerPackage.IsUnitTesting)
+            {
+#pragma warning disable VSTHRD108 // Assert thread affinity unconditionally
+                ThreadHelper.ThrowIfNotOnUIThread();
+#pragma warning restore VSTHRD108
```

Comment:
This doesn't tell me why. Also, comments should end with a period.

Reply:
pls see my latest comment